server/api/operator.go

```diff
@@ -337,18 +338,36 @@ func (h *operatorHandler) Delete(w http.ResponseWriter, r *http.Request) {
 	h.r.JSON(w, http.StatusOK, "The pending operator is canceled.")
 }
 
-func parseStoreIDs(v interface{}) (map[uint64]struct{}, bool) {
-	items, ok := v.([]interface{})
+func parseStoreIDsAndPeerRole(ids interface{}, roles interface{}) (map[uint64]placement.PeerRoleType, bool) {
+	items, ok := ids.([]interface{})
 	if !ok {
 		return nil, false
 	}
-	ids := make(map[uint64]struct{})
+	storeIDs := make([]uint64, 0, len(items))
 	for _, item := range items {
 		id, ok := item.(float64)
 		if !ok {
 			return nil, false
 		}
-		ids[uint64(id)] = struct{}{}
+		storeIDs = append(storeIDs, uint64(id))
 	}
-	return ids, true
+
+	storeIDToPeerRole := make(map[uint64]placement.PeerRoleType)
+
+	peerRoles, ok := roles.([]interface{})
+	if ok && len(peerRoles) >= len(storeIDs) {
```

Comment:
I think it would be better to accept only zero length or same length as storeIDs.

Reply:
Good point.